docs/operations/api-reference.md

```diff
@@ -752,7 +752,11 @@ Returns segment information lists including server locations for the given datas
 
 * `/druid/broker/v1/loadstatus`
 
-Returns a flag indicating if the Broker knows about all segments in Zookeeper. This can be used to know when a Broker process is ready to be queried after a restart.
+Returns a flag indicating if the Broker knows about all segments in the cluster. This can be used to know when a Broker process is ready to be queried after a restart.
+
+* `/druid/broker/v1/readiness`
```

Comment:
I think we should have a more general status endpoint instead of having a specific endpoint for 'readiness'. The status endpoint should return a JSON object. Otherwise, this is taking us down a path where we will have API endpoints for everything we want returned from Druid services and things will get messy and complex fast.

Reply:
I also think 'readiness' is the incorrect term. I think 'available' is stronger.

Reply:
readiness is a better term IMHO because it is an official term for health checking that also has well defined semantics:

https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

More and more apps are exposing health check endpoints to allow for easy integration into orhestration solutions like Kubernetes

Reply:
`/readiness` already exists for historicals so I just extended it for brokers as well, the main reason for its existence is to know when the node is ready to serve queries as opposed to `/status` which just tells if process is up or not. Because of this reason I think we need two endpoints - one for checking when the process is ready to serve and another for checking if the process if up or not. Many monitoring systems rely on HTTP response codes to make decisions about service availability/readiness that's why I think `/readiness` was added even though `/loadstatus` already existed. BTW for health also we have two endpoints `/status` and `/status/health` introduced in https://github.com/apache/incubator-druid/pull/5087.

Reply:
agree with @sascha-coenen readiness is a commonly used term for health checks. 
One change though would be to have a generic endpoint */readiness* instead of node type specific endpoints. 
would suggest removing /druid/<node_type>/v1 prefix.

Reply:
Historical already has `/druid/historical/v1/readiness` so just extended it for broker - `/druid/broker/v1/readiness` to be consistent, but can change it to just `/readiness` if required, no preference from my side.

Reply:
@fjy I am not very concerned about what should be the name of the endpoint, my opinion is that we need an endpoint that returns meaningful HTTP status code so that integration with monitoring/orchestration solutions becomes easier. So I think either we can expose just `/readiness` for all node types that have initialization delay or just add `/druid/broker/v1/readiness` to broker as `/druid/historical/v1/readiness` already exists for historicals. What do you think ?

Reply:
I think 'readiness' is the correct term for this, and putting it at `/druid/broker/v1/readiness` seems acceptable for me since there is precedent with historicals. A universal `/readiness` I think might take a bigger refactor since there isn't really a universal concept of readiness across all node types, or any sort of health style interface that all node types implement, but is worth considering in the future.